test/single_layer/test_convlayer_onednn.cpp

```diff
@@ -0,0 +1,606 @@
+  std::vector<float> kernel_data(3 * 1 * 3 * 3, 1.0f);
+
+  Tensor input = make_tensor(input_data, Shape({1, 3, 5, 5}));
+  Tensor kernel = make_tensor(kernel_data, Shape({3, 1, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel, Tensor(), 3, false, true);
+
+  Tensor output;
+  std::vector<Tensor> in{input};
+  std::vector<Tensor> out{output};
+
+  EXPECT_NO_THROW(layer.run(in, out));
+
+  Shape output_shape = out[0].get_shape();
+  EXPECT_EQ(output_shape[1], 3);
+
+  auto output_data = *out[0].as<float>();
+  for (float val : output_data) {
+    EXPECT_NEAR(val, 9.0f, 1e-5);
+  }
+}
+
+TEST(convlayer_onednn, invalid_input_tensors) {
+  std::vector<float> kernel_data(3 * 3, 1.0f);
+  Tensor kernel = make_tensor(kernel_data, Shape({1, 1, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel);
+  Tensor input1 =
+      make_tensor<float>(std::vector<float>(16, 1.0f), Shape({1, 1, 4, 4}));
+  Tensor input2 =
+      make_tensor<float>(std::vector<float>(16, 1.0f), Shape({1, 1, 4, 4}));
+  Tensor output;
+
+  std::vector<Tensor> in{input1, input2};
+  std::vector<Tensor> out{output};
+
+  EXPECT_THROW(layer.run(in, out), std::runtime_error);
+}
+
+TEST(convlayer_onednn, invalid_input_dimensions) {
+  std::vector<float> kernel_data(3 * 3, 1.0f);
+  Tensor kernel = make_tensor(kernel_data, Shape({1, 1, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel);
+
+  Tensor input =
+      make_tensor<float>(std::vector<float>(4, 1.0f), Shape({1, 2, 2}));
+  Tensor output;
+
+  std::vector<Tensor> in{input};
+  std::vector<Tensor> out{output};
+
+  EXPECT_THROW(layer.run(in, out), std::runtime_error);
+}
+
+TEST(convlayer_onednn, invalid_kernel_dimensions) {
+  std::vector<float> kernel_data(3 * 3, 1.0f);
+  Tensor kernel = make_tensor(kernel_data, Shape({1, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel);
+
+  Tensor input =
+      make_tensor<float>(std::vector<float>(16, 1.0f), Shape({1, 1, 4, 4}));
+  Tensor output;
+
+  std::vector<Tensor> in{input};
+  std::vector<Tensor> out{output};
+
+  EXPECT_THROW(layer.run(in, out), std::runtime_error);
+}
+
+TEST(convlayer_onednn, channel_mismatch_error) {
+  std::vector<float> kernel_data(1 * 2 * 3 * 3, 1.0f);
+  Tensor kernel = make_tensor(kernel_data, Shape({1, 2, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel);
+
+  Tensor input =
+      make_tensor<float>(std::vector<float>(16, 1.0f), Shape({1, 1, 4, 4}));
+  Tensor output;
+
+  std::vector<Tensor> in{input};
+  std::vector<Tensor> out{output};
+
+  EXPECT_THROW(layer.run(in, out), std::runtime_error);
+}
+
+TEST(convlayer_onednn, unsupported_data_type) {
+  std::vector<float> kernel_data(3 * 3, 1.0f);
+  Tensor kernel = make_tensor(kernel_data, Shape({1, 1, 3, 3}));
+
+  ConvLayerOneDnn layer(1, 0, 1, kernel);
+
+  Tensor input =
+      make_tensor<float>(std::vector<float>(16, 1.0f), Shape({1, 1, 4, 4}));
+  Tensor output;
+
+  std::vector<Tensor> in{input};
+  std::vector<Tensor> out{output};
+
+  EXPECT_NO_THROW(layer.run(in, out));
+}
```

Comment:
This test seems to be strange. You check for "unsupported_data_type" but it does not throw. Any mistakes in the checks or test naming?

Reply:
To be honest, even after the change, I don't understand what this test is actually checking